packages/openapi3/src/cli/actions/convert/generators/generate-types.ts

```diff
@@ -47,8 +61,25 @@ function getTypeFromSchema(schema: OpenAPI3Schema): string {
 }
 
 export function getRefName(ref: string): string {
-  const name = ref.split("/").pop() ?? "";
-  return name.split(".").map(printIdentifier).join(".");
+  const { scope, name } = getRefScopeAndName(ref);
+  return [...scope, name].join(".");
+}
+
+export function getRefScopeAndName(ref: string): ReturnType<typeof getScopeAndName> {
+  const parts = ref.split("/");
+  const name = parts.pop() ?? "";
+  const componentType = parts.pop() ?? "";
+  const scopeAndName = getScopeAndName(name);
+
+  switch (componentType) {
+    case "schemas":
+      break;
+    case "parameters":
+      // scopeAndName.scope.unshift("Parameters");
```

Comment:
comment?

Reply:
Woops, this will reappear in a later PR - wanted to break it up as this was already getting pretty large.